packages/shared/src/components/auth/LoginForm.tsx

```diff
@@ -52,28 +61,39 @@ function LoginForm({
     const form = formToJson<LoginFormParams>(formRef.current);
     const { nodes, action } = login.ui;
     const csrfToken = getNodeValue('csrf_token', nodes);
-    const params: LoginPasswordParameters = { ...form, csrf_token: csrfToken };
+    const params: LoginPasswordParameters = {
+      ...form,
+      csrf_token: csrfToken,
+      method: 'password',
+    };
     onPasswordLogin({ action, params });
   };
 
   return (
-    <AuthForm className="gap-2" onSubmit={onLogin} action="#" ref={formRef}>
+    <AuthForm
+      className="gap-2"
+      onSubmit={onLogin}
+      action="#"
+      ref={formRef}
+      data-testid="login_form"
+    >
+      <TokenInput token={getNodeValue('csrf_token', login?.ui?.nodes)} />
       <TextField
         leftIcon={<MailIcon />}
         inputId="identifier"
         name="identifier"
         label="Email"
         type="email"
+        saveHintSpace
+        hint={hint}
+        onChange={() => hint && setHint('')}
```

Comment:
Just checking should it be empty string or null?
Not sure if it renders the UI now